### ISSUE TITLE
Find a way if there's a web-hook to auto update /r/web_design's css with an update from this repo.

### ISSUE THREAD

**Issue by JulianNorton** [JulianNorton/web_design#5]

**Comment by JohnnyDeuss** [comment]
@W3D3 is right about using a hook ([GitHub's Webhooks API](https://developer.github.com/webhooks/)).
I'm willing to code it, but you'll still need a server to host the auto-updating script that gets posted to.
It will also need to have access to git and a moderator account to be able to upload the CSS.

**Comment by JulianNorton** [comment]
@JohnnyDeuss by all means! If you can can it working, I'll figure out how to host it.

**Comment by purduekenny** [comment]
I have a hosting company. I can do it.

**Comment by JohnnyDeuss** [comment]
I've got the script working, it automatically updates the stylesheet and syncs the assets folder with the reddit image.
I am testing with the css/assets from this repo, but I'm getting:

> [line 1823] no image found with name "noresults"
>  						background: transparent url(%%noresults%%) no-repeat 50% 50%;

Am I correct in assuming that noresults is not in the repo and was added manually?

**Comment by JulianNorton** [comment]
@JohnnyDeuss
from: https://github.com/Axel--/Naut-for-reddit/tree/master/Images
I think it's a mistake that we didn't include it in our fork.

**Comment by JohnnyDeuss** [comment]
Seems to work properly. But it does looks like the assets on reddit are not the same as those on GitHub.
This is what I get when I sync everything to [my test subreddit](https://www.reddit.com/r/TEST_WEB_DESIGN/).

**Comment by purduekenny** [comment]
Do I need to do anything still?

**Comment by JohnnyDeuss** [comment]
@purduekenny The script is already on a GitHub repo of mine, so you can already set it up on a server.
The images on this repo are outdated though, or at least not the same as the ones on reddit, so they should be updated here. But as far as the script is concerned, only new/modified images are synced, so you're fine without updating the images for now.
Should I make a pull request or just leave it on my own repo?

**Comment by purduekenny** [comment]
I think a pull request is a good idea. Your thoughts Julian?

**Comment by JulianNorton** [comment]
pull away!

**Comment by purduekenny** [comment]
@JohnnyDeuss I'm not getting it to work. I'm probably doing something wrong, but I'm not sure what. Would you want FTP access to the server I'm putting it on so you can set it up and then I go in and update it with my mod credentials? 

Also, what's your reddit username? I'd like to give you credit for your work.

**Comment by JohnnyDeuss** [comment]
I could definitely do it, but the problem is that I would both need access to the GitHub repo to set up the webhook and access to a reddit account to set up script authorization.
If you want to test what's wrong, you should check the page for that webhook, because GitHub keeps track of all POST data sent to a hook and the reply it gets. The script should be returning some useful error messages that may give a hint as to what is wrong.

**Comment by JohnnyDeuss** [comment]
Actually, if you give me FTP access I can probably pinpoint what is wrong.

**Comment by JulianNorton** [comment]
@purduekenny what's the status on this?

**Comment by purduekenny** [comment]
I flaked. Will follow up today via PM on reddit.

**Comment by purduekenny** [comment]
Sent @JohnnyDeuss a PM with the information to the server. From what I understand, it's in his hands now until I need to add my credentials to it.

**Comment by JohnnyDeuss** [comment]
Looking at the error log it's immediately clear that the script didn't work on the server because it used the <=> from PHP 7, while the server is running 5.6.20, easy fix.
After that, it stills needs access to the git command, which is now resulting in a "command not found" error. Tried to see if I could get it to install on my own, but failed, so A+ on security.

I don't know if you've looked at the comment block at the top of config.php, as it explains what you need to do to get it to work, like having access to git.

**Comment by purduekenny** [comment]
I guess I"m a little dense. I haven't used reddit or github API before. I did read through your comments (great job by the way - I'm probably using it as an example for my team in the coming months), but I took the secret as an optional field and I didn't see any other way to give it access to git - so I assumed it was publicly available. 

I did go through the reddit section and create a reddit app for it, provided a token and a key along with my username and password.

Also, I updated PHP to 7.

**Comment by Haroenv** [comment]
Related to this, wouldn't it be better to store the raw version of the css here, and minimise the css in a build step? (also related to #57 not being able to properly see the diff)

**Comment by JulianNorton** [comment]
Yes, minifying css something us owners can do manually if necessary.
